contracts/extensions/OneTxPayment.sol

```diff
@@ -53,6 +59,59 @@ contract OneTxPayment {
     uint256 _amount,
     uint256 _domainId,
     uint256 _skillId) public
+  {
+    // Check caller is able to call {add,finalize}Payment and moveFundsBetweenPots on the colony in the domain in question
+    validateCallerPermissions(_callerPermissionDomainId, _callerChildSkillIndex, _domainId);
+    // In addition, check the caller is able to call moveFundsBetweenPots from the root domain
+    require(
+      ColonyAuthority(colony.authority()).canCall(msg.sender, 1, address(colony), MOVE_FUNDS_SIG),
+      "colony-one-tx-payment-root-funding-not-authorized"
+    );
+
+    // Add a new payment
+    uint256 paymentId = colony.addPayment(_permissionDomainId, _childSkillIndex, _worker, _token, _amount, _domainId, _skillId);
+    ColonyDataTypes.Payment memory payment = colony.getPayment(paymentId);
+
+    // Fund the payment
+    colony.moveFundsBetweenPots(
+      1, // Root domain always 1
+      0, // Not used, this extension contract must have funding permission in the root for this function to work
+      0, // Not used, this extension contract must have funding permission in the root for this function to work
```

Comment:
Shouldn't this refer to the domainId of the payment? I.e. still be `_childSkillIndex`?